spec/unit/provider/service/launchd_spec.rb

```diff
@@ -343,4 +343,23 @@
       expect { provider.jobsearch("NOSUCH") }.to raise_error(Puppet::Error)
     end
   end
+
+  describe "read_overrides" do
+    before do
+      Kernel.stubs(:sleep)
+    end
+    it "should read overrides" do
+      provider.stubs(:read_plist).returns({})
+      expect(provider.read_overrides).to eq({})
+    end
+    it "should retry if read_plist fails" do
+      provider.stubs(:read_plist).returns(nil)
+      provider.stubs(:read_plist).returns({})
```

Comment:
This isn't gonna do what you're expecting - the second `stubs` will override the first one entirely. I think it will work correctly if you use `expects`, although it runs the expectations in reverse order. You'd want something like:

```
provider.expects(:read_plist).once.returns({})
provider.expects(:read_plist).once.returns(nil)
```

Reply:
Sure! I barely understand rspec. I changed the other tests to use `expects` as well, and they all pass. Also, they fail if I comment out the `.returns({})` or I change `.times(20)` to `.times(19)`